### ISSUE TITLE
Integrate reportportal with XCUItests 

### ISSUE THREAD

**Issue by shiva1810** [reportportal/agent-swift-XCTest#10]
I had all the parameters mentioned in info.plist and did the target settings for the same updated the info.plist file path. I am not seeing anything in reportportal after running tests. if i override the **testcasewillstart** and bundle - then i can see the integration is successful then it is failing at **testsuiteid** not found and expecting all the **rplistener** functions to override. do we have any sample implementation to take a look

**Comment by YauheniPo** [comment]
We integrated the agent with XCUItests for e2e UI autotests by the repository description

**Comment by YauheniPo** [comment]
<img width="738" alt="image" src="https://github.com/user-attachments/assets/ecc4c2a8-f1e5-47dc-9ff0-c3caa1e1f942">

**Comment by YauheniPo** [comment]
We improved and fixed the agent here https://github.com/reportportal/agent-swift-XCTest/pull/9

**Comment by shiva1810** [comment]
I am looking for “/log” endpoint to post both log and screenshot to the
report portal , for now the reporter error method only accept log message ,
if you can enhance this method to post screenshot or attachment to
attachment tab that would be great

Regards,
Shiva M

**Comment by YauheniPo** [comment]
Yes
we have in plans to improve the agent.

**Comment by rusel95** [comment]
here is a request for sending attachments https://github.com/reportportal/agent-swift-XCTest/pull/16

**Comment by rusel95** [comment]
also, reread the fixed Readme file for cocoapods and SPM installation - the Principal class is different there

**Comment by HardNorth** [comment]
Yeah, unfortunately Readme in cocoapods is not updated for some reasons.
